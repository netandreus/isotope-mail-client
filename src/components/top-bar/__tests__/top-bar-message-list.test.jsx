import React from 'react';
import {shallow} from 'enzyme/build/index';
import {TopBarMessageList} from '../top-bar-message-list';

describe('TopBarMessageList component test suite', () => {
  test('Snapshot render, with selected messages and all unread, should render top bar message list', () => {
    // Given
    const props = {t: jest.fn(arg => arg), collapsed: true, title: 'sideshow bob', sideBarToggle: jest.fn(),
      onDeleteClick: jest.fn(), onMarkReadClick: jest.fn(), onMarkUnreadClick: jest.fn(),
      selectedMessages: [{}], selectedMessagesAllUnread: true};
    // When
    const topBarMessageList = shallow(<TopBarMessageList {...props}/>);
    // Then
    expect(topBarMessageList).toMatchSnapshot();
  });
  test('Snapshot render, with selected messages and NOT all unread, should render top bar message list', () => {
    // Given
    const props = {t: jest.fn(arg => arg), collapsed: true, title: 'sideshow bob', sideBarToggle: jest.fn(),
      onDeleteClick: jest.fn(), onMarkReadClick: jest.fn(), onMarkUnreadClick: jest.fn(),
      selectedMessages: [{}], selectedMessagesAllUnread: false};
    // When
    const topBarMessageList = shallow(<TopBarMessageList {...props}/>);
    // Then
    expect(topBarMessageList).toMatchSnapshot();
  });
  test('Snapshot render, with empty selected messages and not all unread, should render top bar message list', () => {
    // Given
    const props = {t: jest.fn(arg => arg), collapsed: true, title: 'sideshow bob', sideBarToggle: jest.fn(),
      onDeleteClick: jest.fn(), onMarkReadClick: jest.fn(), onMarkUnreadClick: jest.fn(),
      selectedMessages: [], selectedMessagesAllUnread: false};
    // When
    const topBarMessageList = shallow(<TopBarMessageList {...props}/>);
    // Then
    expect(topBarMessageList).toMatchSnapshot();
  });
});
