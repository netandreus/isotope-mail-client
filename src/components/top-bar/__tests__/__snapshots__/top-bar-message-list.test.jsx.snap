// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBarMessageList component test suite Snapshot render, with empty selected messages and not all unread, should render top bar message list 1`] = `
<div
  className="mdc-top-app-bar__row"
>
  <section
    className="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
  >
    <Translate(ButtonCollapse)
      collapsed={true}
      sideBarToggle={[MockFunction]}
    />
    <span
      className="mdc-top-app-bar__title"
    >
      sideshow bob
    </span>
  </section>
  <section
    className="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
  >
    <Connect(Translate(ButtonFilter)) />
  </section>
</div>
`;

exports[`TopBarMessageList component test suite Snapshot render, with selected messages and NOT all unread, should render top bar message list 1`] = `
<div
  className="mdc-top-app-bar__row"
>
  <section
    className="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
  >
    <Translate(ButtonCollapse)
      collapsed={true}
      sideBarToggle={[MockFunction]}
    />
    <span
      className="mdc-top-app-bar__title"
    >
      sideshow bob
    </span>
  </section>
  <section
    className="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
  >
    <span
      isotip="topBar.delete"
      isotip-position="bottom"
      isotip-size="small"
    >
      <TopBarButton
        className=""
        onClick={[MockFunction]}
      >
        delete
      </TopBarButton>
    </span>
    <span
      isotip="topBar.markUnread"
      isotip-position="bottom"
      isotip-size="small"
    >
      <TopBarButton
        className=""
        onClick={[MockFunction]}
      >
        markunread
      </TopBarButton>
    </span>
    <Connect(Translate(ButtonFilter)) />
  </section>
</div>
`;

exports[`TopBarMessageList component test suite Snapshot render, with selected messages and all unread, should render top bar message list 1`] = `
<div
  className="mdc-top-app-bar__row"
>
  <section
    className="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
  >
    <Translate(ButtonCollapse)
      collapsed={true}
      sideBarToggle={[MockFunction]}
    />
    <span
      className="mdc-top-app-bar__title"
    >
      sideshow bob
    </span>
  </section>
  <section
    className="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
  >
    <span
      isotip="topBar.delete"
      isotip-position="bottom"
      isotip-size="small"
    >
      <TopBarButton
        className=""
        onClick={[MockFunction]}
      >
        delete
      </TopBarButton>
    </span>
    <span
      isotip="topBar.markRead"
      isotip-position="bottom"
      isotip-size="small"
    >
      <TopBarButton
        className=""
        onClick={[MockFunction]}
      >
        drafts
      </TopBarButton>
    </span>
    <Connect(Translate(ButtonFilter)) />
  </section>
</div>
`;
